api/server/server.go

```diff
@@ -1290,27 +1306,28 @@ func createRouter(eng *engine.Engine, logging, enableCors bool, dockerVersion st
 			"/exec/{id:.*}/json":              getExecByID,
 		},
 		"POST": {
-			"/auth":                         postAuth,
-			"/commit":                       postCommit,
-			"/build":                        postBuild,
-			"/images/create":                postImagesCreate,
-			"/images/load":                  postImagesLoad,
-			"/images/{name:.*}/push":        postImagesPush,
-			"/images/{name:.*}/tag":         postImagesTag,
-			"/containers/create":            postContainersCreate,
-			"/containers/{name:.*}/kill":    postContainersKill,
-			"/containers/{name:.*}/pause":   postContainersPause,
-			"/containers/{name:.*}/unpause": postContainersUnpause,
-			"/containers/{name:.*}/restart": postContainersRestart,
-			"/containers/{name:.*}/start":   postContainersStart,
-			"/containers/{name:.*}/stop":    postContainersStop,
-			"/containers/{name:.*}/wait":    postContainersWait,
-			"/containers/{name:.*}/resize":  postContainersResize,
-			"/containers/{name:.*}/attach":  postContainersAttach,
-			"/containers/{name:.*}/copy":    postContainersCopy,
-			"/containers/{name:.*}/exec":    postContainerExecCreate,
-			"/exec/{name:.*}/start":         postContainerExecStart,
-			"/exec/{name:.*}/resize":        postContainerExecResize,
+			"/auth":                                     postAuth,
+			"/commit":                                   postCommit,
+			"/build":                                    postBuild,
+			"/images/create":                            postImagesCreate,
+			"/images/load":                              postImagesLoad,
+			"/images/{name:.*}/push":                    postImagesPush,
+			"/images/{name:.*}/tag":                     postImagesTag,
+			"/containers/create":                        postContainersCreate,
+			"/containers/{name:.*}/kill":                postContainersKill,
+			"/containers/{name:.*}/pause":               postContainersPause,
+			"/containers/{name:.*}/unpause":             postContainersUnpause,
+			"/containers/{name:.*}/restart":             postContainersRestart,
+			"/containers/{name:.*}/start":               postContainersStart,
+			"/containers/{name:.*}/stop":                postContainersStop,
+			"/containers/{name:.*}/wait":                postContainersWait,
+			"/containers/{name:.*}/resize":              postContainersResize,
+			"/containers/{name:.*}/attach":              postContainersAttach,
+			"/containers/{name:.*}/copy":                postContainersCopy,
+			"/containers/{name:.*}/exec":                postContainerExecCreate,
+			"/exec/{name:.*}/start":                     postContainerExecStart,
+			"/exec/{name:.*}/resize":                    postContainerExecResize,
+			"/containers/{name:.*}/rename/{newname:.*}": postContainerRename,
```

Comment:
I agree that PUT is proper as we are modifying a specific resource. Converting this to a query param is easy but I am not sure if we should make it modify? I love the idea that a single endpoint be provided to modify several attributes of a container but most of the attributes are not persistent.  This PR is to change the name and should remain as rename but again part of me thinks that we should have single endpoint? Please let me know.